src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -1253,6 +1254,9 @@ class BookingService(
             cancellationReason = reason.name,
           ),
         ),
+        metadata = mapOf(
+          MetaDataName.CAS1_CANCELLATION_ID to booking.cancellations.first().id.toString(),
```

Comment:
we should have direct access to the cancellation entity in this function, can we derive the id from that instead please for clarity?

Reply:
Yep i think pass the CancellationEntity straight into createCas1CancellationDomainEvent and then get the id from that. Sorry, i should have checked, i forgot this was in it's own private function